src/randgraphs.jl

```diff
@@ -193,6 +193,191 @@ function barabasi_albert(n::Integer, k::Integer; is_directed=false, seed::Int=-1
     return g
 end
 
+"""
+    static_fitness_model{T<:Real}(m::Int, fitness::Vector{T}; seed::Int=-1)
+
+Generates a non-growing random undirected graph with `length(fitness)` nodes and `m` edges, 
+in which the edge probabilities proportional to node fitness scores. 
+Time complexity is O(|V| + |E| log |E|).
+
+Reference:
+
+* Goh K-I, Kahng B, Kim D: Universal behaviour of load distribution 
+in scale-free networks. Phys Rev Lett 87(27):278701, 2001.
+"""
+function static_fitness_model{T<:Real}(m::Int, fitness::Vector{T}; seed::Int=-1)
+    @assert(m >= 0, "invalid number of edges")
+    n = length(fitness)
+    if m == 0
+        return Graph(n)
+    end
+    # sanity check for the fitness
+    @assert(minimum(fitness) >= zero(eltype(fitness)), "fitness scores must be non-negative")
+    # avoid getting into an infinite loop when too many edges are requested
+    nodes = 0
+    for f in fitness
+        if f > zero(f)
+            nodes += 1
+        end
+    end
+    max_no_of_edges = div(nodes*(nodes-1),2)
+    @assert(m <= max_no_of_edges, "too many edges requested")
+    # calculate the cumulative fitness scores
+    cum_fitness = cumsum(fitness)
+    max_fitness = cum_fitness[end]
+    edges = Set{Edge}()
+    rng = getRNG(seed)
+    while m > 0
+        source = searchsortedfirst(cum_fitness, rand(rng)*max_fitness)
+        target = searchsortedfirst(cum_fitness, rand(rng)*max_fitness)
+        # skip if loop edge
+        (source == target) && continue
+        if source > target
+            source, target = target, source
+        end
+        edge = Edge(source, target)
+        # is there already an edge? If so, try again
+        in(edge, edges) && continue
+        # insert the edge
+        push!(edges, edge)
```

Comment:
`edges(g)` is not efficient, so no - I don't suggest that. What I suggest is keeping the accumulator `m` but do something like this (might be bugs here):

```
g = Graph(n)
while m > 0
  source = searchsortedfirst(cum_fitness, rand(rng)*max_fitness)
  target = searchsortedfirst(cum_fitness, rand(rng)*max_fitness)
  (source == target) && continue
  edge = Edge(source, target)
  has_edge(g, edge) && continue  # don't need to worry about explicit ordering since this is an undirected graph
  add_edge!(g, edge)                     # has_edge and add_edge! will do the right thing
  m -= 1
end
```